provider/terraform.rb

```diff
@@ -135,62 +136,68 @@ def titlelize_property(property)
-      # product_name = data.product.name.underscore
-      # filepath = File.join(target_folder, "resource_#{product_name}_#{name}.go")
-
-      # data.generate('templates/terraform/resource.erb', filepath, self)
-      # generate_documentation(data)
+      # Multi Cloud Swith
+      if $cloud == 'MAC'
+        azure_generate_resource data
+      else
+        dir = data.version == 'beta' ? 'google-beta' : 'google'
+        target_folder = File.join(data.output_folder, dir)
+
+        name = data.object.name.underscore
+        product_name = data.product.name.underscore
+        filepath = File.join(target_folder, "resource_#{product_name}_#{name}.go")
+
+        data.generate('templates/terraform/resource.erb', filepath, self)
+        generate_documentation(data)
+      end
     end
 
     def generate_documentation(data)
-      # TODO: Azure switch
-      azure_generate_documentation data
-
-      # target_folder = data.output_folder
-      # target_folder = File.join(target_folder, 'website', 'docs', 'r')
-      # FileUtils.mkpath target_folder
-      # name = data.object.name.underscore
-      # product_name = data.product.name.underscore
-
-      # filepath =
-      #   File.join(target_folder, "#{product_name}_#{name}.html.markdown")
-      # data.generate('templates/terraform/resource.html.markdown.erb', filepath, self)
+      # Multi Cloud Swith
+      if $cloud == 'MAC'
+        azure_generate_documentation data
+      else
+        target_folder = data.output_folder
+        target_folder = File.join(target_folder, 'website', 'docs', 'r')
+        FileUtils.mkpath target_folder
+        name = data.object.name.underscore
+        product_name = data.product.name.underscore
+
+        filepath =
+          File.join(target_folder, "#{product_name}_#{name}.html.markdown")
+        data.generate('templates/terraform/resource.html.markdown.erb', filepath, self)
+      end
     end
 
     def generate_resource_tests(data)
-      # TODO: Azure switch
-      azure_generate_resource_tests data
-
-      # return if data.object.examples
-      #               .reject(&:skip_test)
-      #               .reject do |e|
-      #             @api.version_obj_or_default(data.version) \
-      #           < @api.version_obj_or_default(e.min_version)
-      #           end
-      #               .empty?
-
-      # dir = data.version == 'beta' ? 'google-beta' : 'google'
-      # target_folder = File.join(data.output_folder, dir)
-
-      # name = data.object.name.underscore
-      # product_name = data.product.name.underscore
-      # filepath =
-      #   File.join(
-      #     target_folder,
-      #     "resource_#{product_name}_#{name}_generated_test.go"
-      #   )
-
-      # data.product = data.product.name
-      # data.resource_name = data.object.name.camelize(:upper)
-      # data.generate('templates/terraform/examples/base_configs/test_file.go.erb',
-      #               filepath, self)
+      # Multi Cloud Swith
+      if $cloud == 'MAC'
```

Comment:
Similar comment as before. We might be able to minimize the changes to one line.

Reply:
Similarly, resolved.